lib/documents.js

```diff
@@ -2702,7 +2654,7 @@ function collectDocUris(forestName, jobState){
     operation.timestamp = new mlutil.Timestamp(null);
     operation.jobState = jobState;
   }
-  requestOptions.path = mlutil.databaseParam(connectionParams, forestEndpoint);
+  requestOptions.path = mlutil.newRequestOptions(connectionParams, forestEndpoint).path;
```

Comment:
This is pretty tricky to make sense of - it raises the question of "Why are we calling `newRequestOptions` but only using the `path`?" 

Doesn't need to be changed now, but it looks like this `collectDocUris` method can be simplified and made more readable by doing the following:

1. Determine `forestEndpoint` as early as possible. It looks like it can be determined based on the `jobState`. 
2. Call `newRequestOptions` as late as possible (right before the `Operation` is instantiated), passing in `forestEndpoint` as the `path` param. 

That should result in a simpler flow:

1. The `forestEndpoint` is calculated.
2. `newRequestOptions` is invoked. 
3. The `Operation` is instantiated. 
4. The `operation` instance is modified as needed. 
5. `startRequest` is invoked. 

You could try doing that here, but I think it's fine to tackle in a separate refactor later on.

Reply:
I tried doing it different ways but since "forestEndpoint" changes based on conditions, this was the only way it was working.